src/main/java/com/epam/reportportal/testng/TestNGService.java

```diff
@@ -87,16 +91,19 @@ public Launch get() {
 				return reportPortal.newLaunch(rq);
 			}
 		});
+		this.stepReporter = StepReporter.getInstance();
 	}
 
 	public TestNGService(Supplier<Launch> launch) {
 		this.launch = new MemoizingSupplier<Launch>(launch);
+		this.stepReporter = StepReporter.getInstance();
 	}
 
 	@Override
 	public void startLaunch() {
 		this.launch.get().start();
 		StepAspect.addLaunch("default", this.launch.get());
+		stepReporter.setLaunch(this.launch.get());
```

Comment:
I believe this call will be done by main thread, so using InheritedThreadLocal in StepReporter.class for launches is just pointless. Please use just simple reference.

Reply:
fixed